### PR TITLE
[YUNIKORN-1744] Enable recovery for MockScheduler

### PR DESCRIPTION
**Pull request by pbacsko**
### What is this PR for?
Remove `IsTestingMode()` calls to make recovery possible when using `MockScheduler`.
Add methods so we can simulate existing nodes and pods.

### What type of PR is it?
* [ ] - Bug Fix
* [x] - Improvement
* [ ] - Feature
* [ ] - Documentation
* [ ] - Hot Fix
* [ ] - Refactoring

### Todos
* [ ] - Task

### What is the Jira issue?
* Open an issue on Jira https://issues.apache.org/jira/browse/YUNIKORN/
* Put link here, and add [YUNIKORN-*Jira number*] in PR title, eg. `[YUNIKORN-2] Gang scheduling interface parameters`

### Questions:
* [ ] - The licenses files need update.
* [ ] - There is breaking changes for older versions.
* [ ] - It needs documentation.

### CODE REVIEW

File: pkg/cache/context_recovery.go [hunk 1]
```diff
@@ -80,86 +73,77 @@ func (ctx *Context) recover(mgr []interfaces.Recoverable, due time.Duration) err
...
-	// we simply simulate to accept or reject nodes on conditions.
-	if !ctx.apiProvider.IsTestingMode() {
-		var podList *corev1.PodList
-		podList, err = ctx.apiProvider.GetAPIs().KubeClient.GetClientSet().
-			CoreV1().Pods("").
-			List(context.Background(), metav1.ListOptions{})
-		if err != nil {
-			return err
-		}
+	pods, err := ctx.apiProvider.GetAPIs().PodInformer.Lister().List(labels.Everything())
```
Comment (pbacsko) [inline pkg/cache/context_recovery.go:76]: I rewrote this to `PodInformer.Lister().List()`, documentation says it's faster.
Although at this point, we already have the list of pods from a previous listing, so we can even skip this (perhaps in a separate JIRA?).

File: pkg/appmgmt/appmgmt.go [hunk 2]
```diff
@@ -56,21 +56,14 @@ func NewAMService(amProtocol interfaces.ApplicationManagementProtocol,
 	}
 
 	log.Logger().Info("Initializing new AppMgmt service")
-
-	if !apiProvider.IsTestingMode() {
-		log.Logger().Info("Registering Spark operator with the AppMgmt service")
```
Comment (zhuqi-lucas) [inline pkg/appmgmt/appmgmt.go:61]: log.Logger().Info("Registering Spark operator with the AppMgmt service")
If this log don't need for new code?

Comment (pbacsko) [reply pkg/appmgmt/appmgmt.go:61]: That line does not add too much information, we always register the Spark operator.

**Review by craigcondit — APPROVED** [review]
+1 LGTM.

**Pull request closed without merging**